mercantile/__init__.py

```diff
@@ -133,7 +133,20 @@ def _parse_tile_arg(*args):
     if len(args) == 1:
         args = args[0]
     if len(args) == 3:
-        return Tile(*args)
+        tile = Tile(*args)
+
+        xtile, ytile, zoom = tile
+
+        if zoom != int(zoom) or not int(zoom) >= 0:
+            raise TileArgParsingError("zoom must be a positive integer")
+
+        if xtile != int(xtile) or not (0 <= int(xtile) <= (2 ** zoom - 1)):
+            raise TileArgParsingError("x must be an integer in [0, 2 ** z-1]")
+
+        if ytile != int(ytile) or not (0 <= int(ytile) <= (2 ** zoom - 1)):
+            raise TileArgParsingError("x must be an integer in [0, 2 ** z-1]")
+
+        return tile
```

Comment:
Because Tile is a namedtuple it's hard to check pre-conditions when it gets constructed.

I thought about changing Tile to a proper class and checking pre-conditions in its constructor, but then I've seen we're using this tile parsing function everywhere before we work on the tile. So this seemed to be a good first step.

Changing Tile to a class and checking pre-conditions in the constructor might still be a good idea to make sure users can not create invalid Tile instances at all: invalid state should not be representable.